Engine/Settings.cs

```diff
@@ -690,40 +690,42 @@ internal static SettingsMode FindSettingsMode(object settings, string path, out
             }
             else
             {
-                var settingsFilePath = settingsFound as String;
-                if (settingsFilePath != null)
-                {
-                    if (IsBuiltinSettingPreset(settingsFilePath))
-                    {
-                        settingsMode = SettingsMode.Preset;
-                        settingsFound = GetSettingPresetFilePath(settingsFilePath);
-                    }
-                    else
-                    {
-                        settingsMode = SettingsMode.File;
-                        settingsFound = settingsFilePath;
-                    }
-                }
-                else
+                if (!TryResolveSettingForStringType(settingsFound, ref settingsMode, ref settingsFound))
                 {
                     if (settingsFound is Hashtable)
                     {
                         settingsMode = SettingsMode.Hashtable;
                     }
-                    else // if the provided argument is wrapped in an expressions then PowerShell resolves it but it will be of type PSObject and we have to operate then on the BaseObject
+                    // if the provided argument is wrapped in an expressions then PowerShell resolves it but it will be of type PSObject and we have to operate then on the BaseObject
+                    else if (settingsFound is PSObject settingsFoundPSObject)
                     {
-                        if (settingsFound is PSObject settingsFoundPSObject)
-                        {
-                            if (settingsFoundPSObject.BaseObject is String)
-                            {
-                                settingsMode = SettingsMode.File;
-                            }
-                        }
+                        TryResolveSettingForStringType(settingsFoundPSObject.BaseObject, ref settingsMode, ref settingsFound);
```

Comment:
there's no check of the return, does it not matter if we get to this point?

Reply:
The check for the return value is only needed for the first method call. If we still cannot resolve the settings type at this point, then the mode will be `SettingsMode.None` (this is the initial default) and that gets handled later on.